en/docs/develop/restapis/email-template.yml

```diff
@@ -0,0 +1,588 @@
+openapi: 3.0.0
+info:
+  version: "v1-oas3"
+  title: WSO2 Identity Server - Email Template API
+  description: >-
+    This is the RESTful API for managing Email Templates in the WSO2 Identity
+    Server.
+  contact:
+    name: WSO2 Identity Server
+    url: 'https://wso2.com/identity-and-access-management/'
+    email: architecture@wso2.com
+  license:
+    name: Apache 2.0
+    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
+tags:
+  - name: Email Template Types
+    description: An email template type can have multiple email templates for different languages.
+  - name: Email Templates
+    description: An email template of a specific type and a language.
+security:
+  - OAuth2: []
+  - BasicAuth: []
+paths:
+  /email/template-types:
+    get:
+      tags:
+        - Email Template Types
+      summary: Retrieve all the Email Template Types.
+      operationId: getAllEmailTemplateTypes
+      description: |
+        Retrieve all the Email Template Types in the system, with limited details of the email templates.
+
+        <b>Permission required:</b>
+        * /permission/admin/manage/identity/emailmgt/view
+      parameters:
+        - $ref: '#/components/parameters/limitQueryParam'
+        - $ref: '#/components/parameters/offsetQueryParam'
+        - $ref: '#/components/parameters/sortOrderQueryParam'
+        - $ref: '#/components/parameters/sortByQueryParam'
+      responses:
+        '200':
+          description: Search results matching the given criteria.
+          content:
+            'application/json':
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/EmailTemplateTypeWithoutTemplates'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '500':
+          $ref: '#/components/responses/ServerError'
+    post:
+      tags:
+        - Email Template Types
+      summary: Add a new email template type.
+      operationId: addEmailTemplateType
+      description: |
+        Adds a new email template type to the system. An email template type can have any number of email templates.
```

Comment:
In some places the descriptions start with "Adds" or "Retrieves" and in some places it starts with the singular form "Add" or "Retrieve". Shall we change it to one of these in all places for consistency?